src/sage/rings/species.py

```diff
@@ -0,0 +1,2337 @@
+            Permutation Group with generators [(1,2)(3,4)(5,6)(7,8,9,10)]
+            sage: H = PermutationGroup([[(1,2,3,4),(5,6),(7,8),(9,10)]]); H
+            Permutation Group with generators [(1,2,3,4)(5,6)(7,8)(9,10)]
+            sage: a = A(G, {0: [1,2,3,4], 1: [5,6,7,8,9,10]}); a
+            {((1,2,3,4)(5,6)(7,8)(9,10),): ({5, 6, 7, 8}, {1, 2, 3, 4, 9, 10})}
+            sage: b = A(H, {0: [1,2,3,4], 1: [5,6,7,8,9,10]}); b
+            {((1,2,3,4)(5,6)(7,8)(9,10),): ({1, 2, 3, 4}, {5, 6, 7, 8, 9, 10})}
+            sage: c = A(G, {0: [1,2,5,6], 1: [3,4,7,8,9,10]}); c
+            {((1,2,3,4)(5,6)(7,8)(9,10),): ({5, 6, 7, 8}, {1, 2, 3, 4, 9, 10})}
+            sage: a == b
+            False
+            sage: a is c
+            True
+        """
+        # a directly indecomposable group has no fixed points, so we
+        # can use the gap group without worrying about the domain
+        G = C.gap()
+        dompart = [[ZZ(C._domain_to_gap[e]) for e in b] for b in dompart]
+        orbits = sorted(G.Orbits().sage(), key=len, reverse=True)
+        for orbit in orbits:
+            if not any(set(orbit).issubset(b) for b in dompart):
+                o = [C._domain_from_gap[e] for e in orbit]
+                raise ValueError(f"All elements of orbit {o} must have the same sort")
+
+        S = SymmetricGroup(C.degree())
+
+        def new_dis():
+            """
+            Return a representative ``dis`` in the conjugacy
+            class of subgroups conjugate to ``C``, together with the
+            conjugating map ``mp``
+            """
+            # create a group conjugate to `C` whose orbits
+            # consist of contiguous sets of numbers, with larger
+            # orbits having smaller numbers, and compute a small
+            # generating set
+            mp = PermutationGroupElement([e for o in orbits for e in o]).inverse().gap()
+            H = PermutationGroup(G.SmallGeneratingSet().sage())
+            return H.conjugate(mp), mp
+
+        key = G.Size().sage(), tuple([len(o) for o in orbits])
+        S_gap = S.gap()
+        if key in parent._dis_cache:
+            lookup_dis = parent._dis_cache[key]
+            for dis in lookup_dis:
+                mp = libgap.RepresentativeAction(S_gap, G, dis)
+                if mp != GAP_FAIL:
+                    break
+            else:
+                dis, mp = new_dis()
+                lookup_dis.append(dis)
+        else:
+            dis, mp = new_dis()
+            lookup_dis = parent._dis_cache[key] = [dis]
+
+        dompart = [[ZZ(e ** mp) for e in b] for b in dompart]
+        mc = tuple([len(b) for b in dompart])
+        key = mc, dis
+        if key in parent._cache:
+            lookup = parent._cache[key]
+            if parent._arity == 1 and lookup:
+                assert len(lookup) == 1
+                return lookup[0]
+            else:
+                domain = list(chain(*map(sorted, dompart)))
```

Comment:
```suggestion
            domain = list(chain(*map(sorted, dompart)))
```
and similar de-indent.